app/src/cli/compile.rs

```diff
@@ -12,24 +12,47 @@ pub struct Args {
 
 pub async fn exec(cmd: Args) -> miette::Result<()> {
     let mut db = Database::from_path(&cmd.filepath);
-    let uri = db.resolve_path(&cmd.filepath)?;
+
+    generate_ir(&mut db, &cmd.filepath).await?;
+    generate_js(&mut db, &cmd.filepath).await?;
+
+    Ok(())
+}
+
+async fn generate_ir(db: &mut Database, filepath: &Path) -> miette::Result<()> {
+    let uri = db.resolve_path(filepath)?;
     let ir = db.ir(&uri).await.map_err(|err| db.pretty_error(&uri, err))?;
 
     if !Path::new(IR_PATH).exists() {
         fs::create_dir_all(IR_PATH).expect("Failed to create IR directory");
     }
 
-    let ir_path = target_path(&cmd.filepath);
+    let ir_path = target_path(filepath, "ir");
     let mut file = fs::File::create(&ir_path).expect("Failed to create file");
     let cfg = PrintCfg::default();
     ir.print_io(&cfg, &mut file).expect("Failed to print to file");
 
     Ok(())
 }
 
-fn target_path(filepath: &Path) -> PathBuf {
+async fn generate_js(db: &mut Database, filepath: &Path) -> miette::Result<()> {
+    let uri = db.resolve_path(filepath)?;
+
+    if !Path::new(IR_PATH).exists() {
+        fs::create_dir_all(IR_PATH).expect("Failed to create IR directory");
+    }
+
+    let js_path = target_path(filepath, "js");
```

Comment:
Might be confusing to reuse the `IR` path for `JS` artefacts. Maybe it is better to add a `JS_PATH` to the file in the driver which defines paths.